### PR TITLE
Add TTS

### PR DESCRIPTION
**Pull request by Royce551**
Still needs implementations for other platforms

### CODE REVIEW

**Review by vicr123 — CHANGES_REQUESTED** [review]
Looks good but can we get locale support in before this gets merged?

File: BnbnavNetClient/Services/TextToSpeech/WindowsTextToSpeechProvider.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    public WindowsTextToSpeechProvider()
+    {
+        if (!OperatingSystem.IsWindows())
+            throw new PlatformNotSupportedException();
+
+        _speechSynthesizer = new SpeechSynthesizer();
+        _speechSynthesizer.SetOutputToDefaultAudioDevice();
+    }
+
+    public Task SpeakAsync(string text, CultureInfo culture)
```
Comment (vicr123) [inline BnbnavNetClient/Services/TextToSpeech/WindowsTextToSpeechProvider.cs]:
```suggestion
    public Task SpeakAsync(string text)
```

File: BnbnavNetClient/Services/TextToSpeech/WindowsTextToSpeechProvider.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    public WindowsTextToSpeechProvider()
+    {
+        if (!OperatingSystem.IsWindows())
+            throw new PlatformNotSupportedException();
+
+        _speechSynthesizer = new SpeechSynthesizer();
+        _speechSynthesizer.SetOutputToDefaultAudioDevice();
+    }
+
```
Comment (vicr123) [inline BnbnavNetClient/Services/TextToSpeech/WindowsTextToSpeechProvider.cs]:
```suggestion

public CultureInfo Culture { get; set; }

```

We should make the culture a global thing

File: BnbnavNetClient/Services/TextToSpeech/ITextToSpeechProvider.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BnbnavNetClient.Services.TextToSpeech;
+interface ITextToSpeechProvider
+{
+    Task SpeakAsync(string text, CultureInfo culture);
+}
```
Comment (vicr123) [inline BnbnavNetClient/Services/TextToSpeech/ITextToSpeechProvider.cs:15]:
```suggestion
    public CultureInfo Culture { get; set; }
}
```

Make the culture a global thing

File: BnbnavNetClient/Services/TextToSpeech/DummyTextToSpeechProvider.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+using System.Threading.Tasks;
+
+namespace BnbnavNetClient.Services.TextToSpeech;
+class DummyTextToSpeechProvider : ITextToSpeechProvider
+{
+    public Task SpeakAsync(string text, CultureInfo culture)
+    {
+        return Task.CompletedTask;
+    }
+}
```
Comment (vicr123) [inline BnbnavNetClient/Services/TextToSpeech/DummyTextToSpeechProvider.cs:18]:
```suggestion
    public CultureInfo Culture { get; set; }
}
```

Make the culture a global thing

File: BnbnavNetClient/Services/TextToSpeech/DummyTextToSpeechProvider.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (vicr123) [inline BnbnavNetClient/Services/TextToSpeech/DummyTextToSpeechProvider.cs:6]:
```suggestion
using System.Globalization;
using System.Threading.Tasks;
```

Clean up imports

File: BnbnavNetClient/Services/TextToSpeech/AppBuilderExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,26 @@
+﻿using Avalonia;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
```
Comment (vicr123) [inline BnbnavNetClient/Services/TextToSpeech/AppBuilderExtensions.cs]:
```suggestion
```

Clean up imports

File: BnbnavNetClient/Services/TextToSpeech/ITextToSpeechProvider.cs [hunk 7]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
```
Comment (vicr123) [inline BnbnavNetClient/Services/TextToSpeech/ITextToSpeechProvider.cs:7]:
```suggestion
using System.Globalization;
using System.Threading.Tasks;
```

Clean up imports

File: BnbnavNetClient/Services/TextToSpeech/WindowsTextToSpeechProvider.cs [hunk 8]
```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Speech.Synthesis;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (vicr123) [inline BnbnavNetClient/Services/TextToSpeech/WindowsTextToSpeechProvider.cs]:
```suggestion
using System;
using System.Globalization;
using System.Speech.Synthesis;
using System.Threading.Tasks;
```

Clean up imports

File: BnbnavNetClient/Services/TextToSpeech/WindowsTextToSpeechProvider.cs [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+        _speechSynthesizer = new SpeechSynthesizer();
+        _speechSynthesizer.SetOutputToDefaultAudioDevice();
+    }
+
+    public Task SpeakAsync(string text, CultureInfo culture)
+    {
+        if (!OperatingSystem.IsWindows())
+            throw new PlatformNotSupportedException();
+
+        var prompt = new PromptBuilder(culture);
```
Comment (vicr123) [inline BnbnavNetClient/Services/TextToSpeech/WindowsTextToSpeechProvider.cs]:
```suggestion
        var prompt = new PromptBuilder(Culture);
```

Make the culture a global thing

**Review by vicr123 — CHANGES_REQUESTED** [review]
We should make the culture a global thing - it doesn't make sense to constantly pass it to the speech function when the application language only changes on restart.

**Review by Vrabbers — CHANGES_REQUESTED** [review]
Either use a global culture property, or perhaps utilize ``CultureInfo.CurrentUICulture`` directly, as that will have the value of the text in it already

**Review by vicr123 — CHANGES_REQUESTED** [review]
'REQUEST_CHANGES' performed by Xcode.

**Review by vicr123 — APPROVED** [review]
'APPROVE' performed by Xcode.

**Pull request merged**